flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultSubpartitionView.java

```diff
@@ -55,6 +55,14 @@ default void notifyPriorityEvent(int priorityBufferNumber) {}
 
     void acknowledgeAllDataProcessed();
 
+    /**
+     * {@link ResultSubpartitionView} can decide whether the failure cause should be reported to
+     * consumer as failure (primary failure) or as cancellation (secondary failure). Secondary
```

Comment:
Maybe let's rephrase this and replace `as cancellation` with `{@link ProducerFailedException}`?